nixos/doc/manual/release-notes/rl-2311.section.md

```diff
@@ -1017,7 +1017,7 @@ Make sure to also check the many updates in the [Nixpkgs library](#sec-release-2
 
 - [trust-dns](https://trust-dns.org/), a Rust based DNS server built to be safe
   and secure from the ground up. Available as
-  [services.trust-dns](#opt-services.trust-dns.enable).
+  [services.trust-dns](options.html#opt-services.trust-dns.enable).
```

Comment:
the release notes don't seem to handle renamed options: the nixos manual was failing to build unless i change the format to this.

Reply:
Please move this to 24.11 release notes.

Reply:
It should also mention the rename.

Reply:
this change is required for the manual to build. otherwise:
```
$ nix-build nixos/release.nix -A manual.x86_64-linux
...
       > RuntimeError: failed to render manual manual.md
       > error:
       >        failed to render manual manual.md
       >
       > caused by:
       >         rendering release-notes/release-notes.md
       >
       > caused by:
       >  rendering release-notes/rl-2311.section.md
       >
       > caused by:
       > bad local reference, id #opt-services.trust-dns.enable not known
       For full logs, run 'nix log /nix/store/bhmzf4h92cghvjf980vn4ypdjh1k69r9-nixos-manual-html.drv'.
```

this seems to be because `mkRenamedOptionModule` hides the old option name from the docs. if it's absolutely forbidden to edit the old release notes, then the other way to fix this is to use `mkAliasOptionModule` instead, which will keep both `services.hickory-dns.*` and `services.trust-dns.*` visible in the docs.

it's a weird error. would love some advice if you've seen this before.

Reply:
Why are you trying to change the old release notes?

Reply:
This is a diff on the 23.11 release notes. Maybe try removing this change?

Reply:
removed the changes to 23.11 and pushed to this PR. ofborg confirms a failure to build the manual: <https://github.com/NixOS/nixpkgs/actions/runs/9947857257/job/27481347451?pr=316466>